frame/nomination-pools/src/lib.rs

```diff
@@ -1103,7 +1119,55 @@ pub mod pallet {
 				},
 			);
 			bonded_pool.put();
-			Self::deposit_event(Event::<T>::Joined { delegator: who, pool_id, bonded: amount });
+			Self::deposit_event(Event::<T>::Bonded {
+				delegator: who,
+				pool_id,
+				bonded: amount,
+				joined: true,
+			});
+
+			Ok(())
+		}
+
+		/// bond up to `extra` more funds from `origin` into the pool to which they already
+		/// belong.
+		///
+		/// Additional funds can come from either the free balance of the account, of from the
+		/// accumulated rewards, see [`BondExtra`].
+		// NOTE: this transaction is implemented with the sole purpose of readability and
+		// correctness, not optimization. We read/write several storage items multiple times instead
+		// of just once, in the spirit reusing code.
+		#[pallet::weight(0)]
+		#[transactional]
+		pub fn bond_extra(origin: OriginFor<T>, extra: BondExtra<BalanceOf<T>>) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let (mut delegator, mut bonded_pool, mut reward_pool) =
+				Self::get_delegator_with_pools(&who)?;
+
+			let (points_issued, bonded) = match extra {
+				BondExtra::FreeBalance(amount) => {
+					let amount = amount.min(T::Currency::free_balance(&who));
```

Comment:
I don't think this min will work. We always do transfer keep alive for delegators (including in `try_bond_funds`), so we can't transfer all of their free balance. IMO we should always just fail if they don't have the amount

Reply:
yeah I also don't really like it, it is magic logic hidden in the code. I was trying to mimic the staking `bond_extra`, which does this. 

If we actually want to `.min()`, we do: 

```suggestion
					let amount = amount.min(T::Currency::free_balance(&who) - ED);
```

Reply:
removed in 123234233651f936af9988746e38bb3d78141daa